python/bro.py

```diff
@@ -49,27 +53,72 @@ def get_binary_stdio(stream):
             return orig_stdio.buffer
 
 
-def main(args):
-
-    options = parse_options(args)
+def main():
+
+    parser = argparse.ArgumentParser(
+        prog='bro.py',
+        description="Compression/decompression utility using the Brotli algorithm.")
+    parser.add_argument('--version', action='version', version='%(prog)s 1.0')
+    parser.add_argument('-i', '--input', metavar='FILE', type=str, dest='infile',
+                        help='Input file', default=None)
+    parser.add_argument('-o', '--output', metavar='FILE', type=str, dest='outfile',
+                        help='Output file', default=None)
+    parser.add_argument('-f', '--force', action='store_true',
+                        help='Overwrite existing output file', default=False)
+    parser.add_argument('-d', '--decompress', action='store_true',
+                        help='Decompress input file', default=False)
+    params = parser.add_argument_group('optional encoder parameters')
+    params.add_argument('-m', '--mode', metavar="MODE", type=int, choices=[0, 1],
+                        help='The compression mode can be 0 (text) or 1 (font). '
+                        'Defaults to text mode.')
+    params.add_argument('-q', '--quality', metavar="QUALITY", type=int,
+                        choices=list(range(0, 12)),
+                        help='Controls the compression-speed vs compression-density '
+                        'tradeoff. The higher the quality, the slower the '
+                        'compression. Range is 0 to 11. Defaults to 11.')
+    params.add_argument('--lgwin', metavar="LGWIN", type=int,
+                        choices=list(range(16, 25)),
+                        help='Base 2 logarithm of the sliding window size. Range is '
+                        '16 to 24. Defaults to 22.')
+    params.add_argument('--lgblock', metavar="LGBLOCK", type=int,
+                        choices=[0] + list(range(16, 25)),
+                        help='Base 2 logarithm of the maximum input block size. '
+                        'Range is 16 to 24. If set to 0, the value will be set based '
+                        'on the quality. Defaults to 0.')
+    above9 = parser.add_argument_group(
+                        'encoder parameters respected only if quality > 9.')
```

Comment:
Those under "// These settings will be respected only if quality > 9."
should be removed.

On Mon, May 11, 2015 at 11:46 AM, Cosimo Lupo <notifications@github.com>
wrote:

> In python/bro.py
> <https://github.com/google/brotli/pull/105#discussion_r30025220>:
>
> > +    params.add_argument('-q', '--quality', metavar="QUALITY", type=int,
> > +                        choices=list(range(0, 12)),
> > +                        help='Controls the compression-speed vs compression-density '
> > +                        'tradeoff. The higher the quality, the slower the '
> > +                        'compression. Range is 0 to 11. Defaults to 11.')
> > +    params.add_argument('--lgwin', metavar="LGWIN", type=int,
> > +                        choices=list(range(16, 25)),
> > +                        help='Base 2 logarithm of the sliding window size. Range is '
> > +                        '16 to 24. Defaults to 22.')
> > +    params.add_argument('--lgblock', metavar="LGBLOCK", type=int,
> > +                        choices=[0] + list(range(16, 25)),
> > +                        help='Base 2 logarithm of the maximum input block size. '
> > +                        'Range is 16 to 24. If set to 0, the value will be set based '
> > +                        'on the quality. Defaults to 0.')
> > +    above9 = parser.add_argument_group(
> > +                        'encoder parameters respected only if quality > 9.')
>
> I see. Which arguments shall I keep or remove precisely?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/brotli/pull/105/files#r30025220>.
>